### DISCUSSION TITLE
k3s-killall and k3s-uninstall not gonna  stop containers nor delete resources

### DISCUSSION THREAD

**Discussion by lidh15** [k3s-io/k3s#6160] [Q&A]
I thought these two scripts helped to eliminate all modifications brought by k3s, but when I tried to restore my machine I found that after running them all the containers are still there.

What should I do if I want to time travel to where I've never met Miss K3s?

**Comment by brandond** [comment]
The uninstall script should remove everything. What in particular are you seeing left on your host?

**Reply by lidh15** [reply]
after running uninstall there was no longer 'kubectl' executable, so I ran 'docker ps' (yes I'm using docker backend and asked many questions about this if you remember) and I see all containers of kube-system are still running.

**Reply by brandond** [reply]
Yes, If you're using an external container runtime, the killall script doesn't clean up the Docker containers started by the Kubelet. There are many potential different container runtimes that might be used with --docker or --container-runtime-endpoint, so we make no attempt to guess at what they might be and handle them all.

**Reply by lidh15** [reply]
Okay I will manage to script it out of k3s. Thank you.
